cEP-0028.md

```diff
@@ -0,0 +1,505 @@
+       """
+       Add a proxy instance to the proxy map.
+
+       :param proxy:
+           The proxy instance to register in the map.
+       :param replace:
+           A boolean flag indicating if the proxy should
+           replace an existing proxy of the same file.
+       :return:
+           Boolean true if registering of the proxy was
+           successful else false.
+       """
+       if not isinstance(proxy, FileProxy):
+         return False
+
+       if self._map.get(proxy.filename) is not None:
+         if replace:
+           self._map[proxy.filename] = proxy
+           return True
+         return False
+
+       self._map[proxy.filename] = proxy
+       return True
+
+     def resolve(self, filename: str, workspace=None, hard_sync=True):
+       """
+       Resolve tries to find an available proxy or creates one
+       if there is no available proxy for the said file.
+
+       :param filename:
+           The filename to search for in the map or to create
+           a proxy instance using.
+       :param workspace:
+           Used in case the lookup fails and a new instance is
+           being initialized.
+       :hard_sync:
+           Boolean flag indicating if the file should be initialized
+           from the file on disk or fail otherwise.
+       :return:
+           Returns a proxy instance or a boolean indicating the
+           failure of the resolution.
+       """
+       proxy = self.get(filename)
+       if proxy is not None:
+         return proxy
+
+       try:
+         proxy = FileProxy.from_name(filename, workspace)
+       except FileNotFoundError:
+         if hard_sync:
+           return False
+
+       try:
+         proxy = FileProxy(filename, workspace)
+       except Exception:
+         return False
+
+       self.add(proxy)
+       return proxy
+   ```
+
+   - `coalib.processes.Processing.get_file_dict()`
+
+     Requires patching `get_file_dict()` to conditionally build
+     file dict from a FileProxyMap instance if available rather
+     than from disk files.
+
+   ```python
+   def get_file_dict(filename_list,
+                     log_printer=None,
+                     allow_raw_files=False,
+                     fileproxy_map=None):
+     """
+     Reads all files into a dictionary.
+     :param filename_list:   List of files to get the contents of.
+     :param log_printer:     The logger which logs errors.
+     :param allow_raw_files: Allow the usage of raw files (non text files),
+                             disabled by default
+     :param fileproxy_map:   The FileProxyMap instance to use while building
+                             a file dict when available.
+     :return:                Reads the content of each file into a dictionary
+                             with filenames as keys.
+     """
+     file_dict = {}
+
+     # If fileproxy_map is available, use it to build the file dict.
+     if fileproxy_map is not None:
```

Comment:
I think this `if` should be inside of the `for`.

most of the logic is the same for both (with a bit of adjustment to each of them).

keeping the branches close together and very small will prevent it being broken by someone only updating one branch.